File: SecondYear/Geek-shreya_ShreyaGarg_2226cse1095_2/Week_1/Pattern_8.py
#User function Template for python3

class Solution:
    def printTriangle(self, N):
        for i in range(1,N+1):
            k=1
            for j in range(1,N+1):
            
                if j<=i:
                    print(k,"",end="")
                    k=k+1
                else:
                    print(" ",end="")
            k=k-1
            for l in range(1,N+1):
                if l>=(N+1-i):
                    print(k,"",end="")
                    k=k-1
                else:
                    print("   ",end="")
                    
            print()
#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input().strip())
        ob = Solution()
        ob.printTriangle(N)
# } Driver Code Ends

File: SecondYear/Junior-Hawk_ArchitAgarwal_2226cseai1186_2/Week_1/pattern8.cpp
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
class Solution {
  public:
    void printTriangle(int n) {
        for(int i=1;i<=n;i++){
            for(int j=1;j<=i;j++){
                cout<<j<<" ";
                //if(j!=i) cout<<" ";
            }
            for(int j=1;j<=(4*(n-i));j++){
                cout<<" ";
            }
            for(int j=i;j>0;j--){
                cout<<j<<" ";
            }
            cout<<endl;
            }
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;

        Solution ob;
        ob.printTriangle(n);
    }
    return 0;
}